autoload/ale/job.vim

```diff
@@ -174,14 +174,20 @@ function! ale#job#PrepareCommand(command) abort
     " but we'll do this explicitly, so we use thes same exact command for both
     " versions.
     if ale#Has('win32')
-        return 'cmd /c ' . a:command
+        if g:ale_cmd_wrapper isnot# ''
+            return g:ale_cmd_wrapper . ' cmd /c ' . a:command
+        else
+            return 'cmd /c ' . a:command
+        endif
     endif
 
+    let l:precmd = split(g:ale_cmd_wrapper)
```

Comment:
What happens when you have a command like `'"/path to script/foo" bar'`? It can be split weirdly.

How should wrappers which want the entire command to be passed as one argument be handled?

Reply:
Yes, that'd split them weirdly and it wouldn't work. But I think the same limitation applies to the `&shell` variable. Is that a problem? Other option is not to split it and support just a command, without arguments. I really don't want to go the whole way of shell-aware splitting. Or maybe the cmd wrapper could be a list, but that feels a bit inconsistent with the rest of ale's settings.

I don't think such wrappers that wants it as a single argument need to be supported. I mean, this is just a tiny feature and probably doesn't need a whole framework of handling of commands, so considering how many (or few) people will use it and that such command can be handled by writing a shell script to convert it is easy, it would be OK to support just a single mode. Or, in other words, postponing the other mode until one actually needs it 😇. Does that sound OK, or did you have some specific (simple) solution in mind?

Reply:
Okay, could you document all of this? Document that the command must not contain spaces, and the command must eat all of the following arguments.

Reply:
I reworded the thing with multiple arguments to be hopefully clearer and added the one with spaces.